packaging/installer/kickstart-ng.sh

```diff
@@ -0,0 +1,843 @@
+#!/bin/sh
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+# ======================================================================
+# Constants
+
+REPOCONFIG_URL_PREFIX="https://packagecloud.io/netdata/netdata-repoconfig/packages"
+REPOCONFIG_VERSION="1-1"
+PATH="${PATH}:/usr/local/bin:/usr/local/sbin"
+
+# ======================================================================
+# Defaults for environment variables
+
+RELEASE_CHANNEL="nightly"
+NETDATA_CLAIM_URL="https://app.netdata.cloud"
+NETDATA_USER_CONFIG_DIR="/etc/netdata"
+NETDATA_AUTO_UPDATES="1"
+NETDATA_ONLY_NATIVE=0
+NETDATA_ONLY_STATIC=0
+
+NETDATA_DISABLE_TELEMETRY="${DO_NOT_TRACK:-0}"
+NETDATA_TARBALL_BASEURL="${NETDATA_TARBALL_BASEURL:-https://storage.googleapis.com/netdata-nightlies}"
+
+if [ ! -t 1 ]; then
+  INTERACTIVE=0
+else
+  INTERACTIVE=1
+fi
+
+# ======================================================================
+# Usage info
+
+usage() {
+  cat << HEREDOC
+USAGE: kickstart.sh [options]
+       where options include:
+
+  --dont-wait                Do not prompt for user input. (default: prompt if there is a controlling terminal)
+  --non-interactive          Do not prompt for user input.
+  --dont-start-it            Do not start the agent by default (only for static installs or local builds)
+  --interactive              Prompt for user input even if there is no controlling terminal.
```

Comment:
The logic here is as follows:

* If none of the options are specified, we decide whether a run is ‘interactive’ based on whether we see a controlling terminal (see the conditional block starting on line 25 for how we do this).
* Otherwise, we follow whichever of the options was the last specified.

This is an improvement over the existing scripts in that it will _usually_ do the right thing (but I’m not stupid enough to believe that it will always do the right thing).

The synonymous nature of the first two options is for compatibility with the existing script (which has both options). The options really just exist in this iteration though to override that auto-detection.

Reply:
Oh, so `--dont-wait` and `--non-iteractive` are _intentionally_ doing the same thing. In that case, let's remove the `--dont-wait` option just from the help text and keep it in the switch statement that handles the script's options. Update the order as well so that `--interactive` and `--non-interactive` are next to each other.

Reply:
Done, marking this resolved.